Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,1517 @@
+                  typename boost::mpl::eval_if<
+                  boost::has_range_iterator<Polyline>,
+                  boost::range_value<Polyline>,
+                  boost::false_type >::type
+                  >::type
+                  >::type* = 0
+                  )
+{
+  CGAL_precondition(CGAL::is_triangle_mesh(mesh));
+
+  return do_intersect(mesh, polyline, parameters::all_default());
+}
+}//end PMP
+namespace internal{
+
+template<class TriangleMeshRange
+         , typename OutputIterator
+         , class NamedParametersRange
+         , class NamedParameter
+         >
+struct Mesh_callback
+{
+  typedef typename boost::range_value<TriangleMeshRange>::type TriangleMesh;
+  typedef typename GetGeomTraits<TriangleMesh, NamedParameter>::type GT;
+  typedef typename boost::range_value<NamedParametersRange>::type VertexPointMapNP;
+  typedef typename GetVertexPointMap<TriangleMesh, VertexPointMapNP>::const_type VPM;
+  typedef CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, VPM> Primitive;
+  typedef CGAL::AABB_traits<GT, Primitive> Traits;
+  typedef CGAL::AABB_tree<Traits> AABBTree;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor face_descriptor;
+
+  // todo : add a vector/map of trees for each mesh.
+  // fill them in the operator and test for inclusion with
+  // Side_of_triangle_mesh.
+  const TriangleMeshRange& meshes;
+  OutputIterator m_iterator;
+  const bool& test_volume;
+  const NamedParametersRange& nps;
+  const NamedParameter& np;
+  std::vector<AABBTree*> trees;
+
+  std::vector<std::vector<CGAL::Point_3<GT> > > points_of_interest;
+
+  Mesh_callback(const TriangleMeshRange& meshes,
+                OutputIterator iterator,
+                const bool& test_volume,
+                const NamedParametersRange& nps,
+                const NamedParameter& np)
+    : meshes(meshes), m_iterator(iterator),
+      test_volume(test_volume), nps(nps), np(np)
+  {
+    int size = std::distance(meshes.begin(), meshes.end());
+    trees = std::vector<AABBTree*>(size, NULL);
+    points_of_interest.resize(size);
+  }
+
+  template<class TriangleMesh
+           , class VPM
+           , class GT
+           >
+
+  bool is_mesh1_in_mesh2(const TriangleMesh& mesh_1,
+                         const TriangleMesh& mesh_2,
+                         const int& mesh_id_1,
+                         const int& mesh_id_2,
+                         const VPM& vpm1,
+                         const VPM& vpm2,
+                         const GT& gt)
+  {
+    //test if mesh2 is included in mesh1
+
+    //get AABB_tree for mesh1
+    if(!trees[mesh_id_1])
+    {
+      trees[mesh_id_1] = new AABBTree(faces(mesh_1).begin(),
+                                      faces(mesh_1).end(),
+                                      mesh_1);
+    }
+    //get a face-index map for mesh2
+    if(points_of_interest[mesh_id_2].size() == 0)
+    {
+      std::map<face_descriptor, int> fid_map;
```

Comment:
`unordered_map` and you can even use the dynamic_property_map mecanism I think.

Reply:
I don't want to force the user to provide a face ibndex map, as it is useless for the non-volumic calculation.

Reply:
this is particularly the goal of dynamic_property_map